Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -455,12 +455,16 @@ - (void)_updateScreenTimeViewGeometry
     [[_screenTimeWebpageController view] setFrame:bounds];
 }
 
-- (void)_updateScreenTimeShieldVisibilityForWindow
+- (void)_updateScreenTimeBasedOnWindowVisibility
 {
     BOOL viewIsInWindow = !!self.window;
-
+    BOOL viewIsVisible = viewIsInWindow;
     BOOL showsSystemScreenTimeBlockingView = [_configuration _showsSystemScreenTimeBlockingView];
 
+#if PLATFORM(MAC)
+    viewIsVisible = viewIsVisible && ((self.window.occlusionState & NSWindowOcclusionStateVisible) == NSWindowOcclusionStateVisible);
```

Comment:
We should keep `viewIsVisible` logic next to each other? Seems weird to have `BOOL showsSystemScreenTimeBlockingView` in the middle.